hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -1227,14 +1303,34 @@ private JobAndSqlSummary getJobAndSqlSummary(JobRecord record) {
         String suspensionCause = executionRecord != null && executionRecord.getSuspensionCause() != null
                 ? executionRecord.getSuspensionCause().description() : null;
         JobStatus status;
+        boolean userCancelled;
         if (ctx == null) {
+            // If we have a JobRecord but not the MasterContext it may mean that:
+            // 1) job has not yet created MasterContext => NOT_RUNNING
+            // 2) job is suspended => SUSPENDED
+            // 3) job has already ended but JobRecord has not yet been deleted =>
+            //    do not care, result will be overwritten by the one obtained from JobResult
+            //    which is guaranteed to exist in this case
             status = executionRecord != null && executionRecord.isSuspended()
                     ? JobStatus.SUSPENDED : JobStatus.NOT_RUNNING;
+            userCancelled = false;
         } else {
-            status = ctx.jobStatus();
+            // order of reads is important, see comment in determineJobStatusFromMasterContext
+            // for consistent result we must use single instance of TerminationRequest for all checks
+            Optional<TerminationRequest> maybeTerminationRequest = ctx.jobContext().getTerminationRequest();
+            JobStatus jobStatus = ctx.jobStatus();
+            status = jobStatus == RUNNING && maybeTerminationRequest.isPresent()
+                    ? COMPLETING
+                    : jobStatus;
```

Comment:
`getJobAndSqlSummary` determines both `status` and `userCancelled`. For both of them we need `jobStatus` and `maybeTerminationRequest`. I cannot simply use `determineJobStatusFromMasterContext` because they would be read twice which could introduce inconsistency in result.